js/block.js

```diff
@@ -853,11 +950,63 @@ function createVariableBlock(initialValue) {
    variableStep
       .querySelector('wb-value[type="any"]')
       .appendChild(initialValue);
+
    // TODO: autogenerate a good name.
 
    return variableStep;
 }
 
 
+
+function setTypeOfVariable(variableStep, type){
+   // Set type of variable to match type of object
+   variableStep
+       .querySelector('[script="control.getVariable"]') // get local expression
+       .setAttribute('type', type);
+}
+
+function updateLocalInstances(variableStep, type){
+    var parentContext = dom.closest(variableStep, 'wb-contains');
+    var variableLocalsToUpdate = getVariablesToUpdate(parentContext, variableStep.id);
+    variableLocalsToUpdate.forEach(function(varinstance){
+        dom.closest(varinstance, 'wb-expression').setAttribute('type', type);
+    });
+}
+
+/* Add custom events for adding and removing blocks from the DOM */
+
```

Comment:
This is a weird bit. It is basically a fix for CustomElement lifecycle "events" not really being events. It almost belongs in `event.js` except that it is only called for block components. If we could move it to events, but initialize it with a list of element names to pay attention to, that would be the best.